riscv-sbi.adoc

```diff
@@ -70,13 +70,44 @@ Standard SBI error codes are listed below
 
 [cols="<,>",options="header,compact"]
 |===
-|  Error Type              |Value
-|  SBI_SUCCESS             |  0
-|  SBI_ERR_FAILURE         | -1
-|  SBI_ERR_NOT_SUPPORTED   | -2
-|  SBI_ERR_INVALID_PARAM   | -3
-|  SBI_ERR_DENIED          | -4
-|  SBI_ERR_INVALID_ADDRESS | -5
+|  Error Type                |Value
+|  SBI_SUCCESS               |  0
+|  SBI_ERR_FAILED            | -1
+|  SBI_ERR_NOT_SUPPORTED     | -2
+|  SBI_ERR_INVALID_PARAM     | -3
+|  SBI_ERR_DENIED            | -4
+|  SBI_ERR_INVALID_ADDRESS   | -5
+|  SBI_ERR_ALREADY_AVAILABLE | -6
+|===
+
+Every SBI function should prefer `unsigned long` as the data type. It keeps the
+specification simple and easily adaptable for all RISC-V ISA types (i.e. RV32,
+RV64 and RV128). In case the data is defined as 32bit wide, the upper 32 bits
+should be considered as reserved bits unless otherwise specified.
```

Comment:
I don't like this: 32-bit types are well defined by the ABI and we should respect that.  If there's a 32-bit type in a function signature then it's probably a 32-bit quantity, so there's no reason to pack other stuff into that register.

Reply:
However, more-privileged software shouldn't trust that less-privileged software has adhered to the ABI.  In particular, it shouldn't assume sub-XLEN types have been properly extended.  So I think the less-privileged software should be required to adhere to the ABI, as Palmer suggests (e.g. 32 bit values always sign-extended in 64-bit registers), but more-privileged software should assume it is passed as unsigned long, then cast it back to uint32 to do the extension.

Reply:
ok. I will update this.